tests/unit/commands/test_runner/test_init_command.py

```diff
@@ -123,10 +128,38 @@ def test_valid_tagging_api_response(self, boto_client_provider_patch, query_tagg
                 template_name=self.TEST_TEMPLATE_NAME,
                 env_file=self.TEST_ENV_FILE_NAME,
                 image_uri="test_image_uri",
+                allow_iam=False,
                 runtime="python3.8",
             )
             self.assertTrue(os.path.exists(self.TEST_TEMPLATE_NAME))
             self.assertTrue(os.path.exists(self.TEST_ENV_FILE_NAME))
         finally:
             os.remove(self.TEST_TEMPLATE_NAME)
             os.remove(self.TEST_ENV_FILE_NAME)
+
+    @patch("samcli.commands.test_runner.init.cli.query_tagging_api")
+    @patch("samcli.lib.utils.boto_utils.get_boto_client_provider_with_config")
+    def test_valid_tagging_api_response_with_allow_iam(self, boto_client_provider_patch, query_tagging_api_patch):
+        mock_ctx = Mock()
+        mock_ctx.region = "test-region"
+        mock_ctx.profile = "test-profile"
+        boto_client_provider_patch.return_value = None
+        query_tagging_api_patch.return_value = ["arn:aws:lambda:us-east-1:123456789123:function:valid-lambda-arn"]
+
+        try:
+            do_cli(
+                ctx=mock_ctx,
+                tags={"Key": "Value"},
+                template_name=self.TEST_TEMPLATE_NAME,
+                env_file=self.TEST_ENV_FILE_NAME,
+                image_uri="test_image_uri",
+                allow_iam=True,
+                runtime="python3.8",
+            )
+            self.assertTrue(os.path.exists(self.TEST_TEMPLATE_NAME))
+            self.assertTrue(os.path.exists(self.TEST_ENV_FILE_NAME))
+
+            self.assertFalse("#" in Path.read_text(Path(self.TEST_TEMPLATE_NAME)))
```

Comment:
This seems brittle if we later add comments to the yaml file; would it be better to assert that the actions do/don't show up if you parse the yaml?

Reply:
Good point, I think so